index.js

```diff
@@ -0,0 +1,158 @@
+//
+// Seed files allow you to populate your database with test or seed data independent of your migration files.
+// yarn knex seed:make seed_name                // makes a seed file called seed_name
+// yarn knex seed:run                           // Runs seed files, populates you db with seed data
+
+
+// Knex configuration. Then, assign it to 'db', invoking the config through knex
+const knexConfig = {
+    client: 'sqlite3',
+    connection: {
+      filename: './data/lambda.sqlite3',
+    },
+    useNullAsDefault: true,             // needed for sqlite
+    // debug: true,                        // Useful for development
+};
+const db = knex(knexConfig);
+
+// Variables
+const port = process.env.PORT || 5000;
+const server = express();
+const errors = { // Dynamic error messaging based on sqlite codes
+    '1': 'We ran into an error.',
+    '4': 'Operation aborted',
+    '9': 'Operation aborted',
+    '19': 'Another record with that value exists, yo!'
+};
+
+// Middleware
+server.use(helmet());
+server.use(express.json());
+server.use('/students', students);
+
+
+// Endpoints
+// GET all cohorts
+server.get('/api/cohorts', async (req, res) => {
+    try {
+        const cohorts = await db('cohorts');
+        res.status(200).json(cohorts);
+    } catch (error) {
+        res.status(500).json(error);
+    }
+});
+
+
+// GET a cohort by id
+server.get('/api/cohorts/:id', async (req, res) => {
+    try {
+        const cohort = await db('cohorts')
+            .where({ id: req.params.id })
+            .first();
+        res.status(200).json(cohort);
+    } catch (error) {
+        res.status(500).json(error);
+    }
+});
+
+
+// GET students in cohort by cohort ID
+server.get('/api/cohorts/:id/students', async (req, res) => {
+    try {
+        const cohortStudents = await db('cohorts')
+            .innerJoin('students', `cohorts.id`, 'students.cohort_id')
+            .where('students.id', req.params.id)
+        if (cohortStudents.length < 1) {
+            return res.status(404).json({ message:"No students in given cohort" })
+        }
+        res.status(200).json(cohortStudents)
+    } catch (error) {
+        res.status(500).json(error)
+    }
+});
+
+
+// POST create new cohort. Name is required.
+server.post('/api/cohorts', async (req, res) => {
+    if (!req.body.name) { return res.status(400).json({ message:"Please include a name to create a new cohort" })}
+    try {
+        const [id] = await db('cohorts').insert(req.body);
+        const cohort = await db('cohorts')
+            .where({ id })
+            .first();
+        res.status(201).json(cohort);
+    } catch (error) {
+        const message = errors[error.errno] || 'We ran into an error';
+        res.status(500).json({ message });
+    }
+});
+
+
+// PUT update without name param
+server.put('/api/cohorts', async (req, res) => {
+    res.status(400).json({ message:"Please include cohort ID to update" })
+})
```

Comment:
instead of writing a whole extra endpoint for this, you could leave it out entirely since Postman will throw it;s own error to you if you try to run a PUT that expects an id when it isn't passed one